### DISCUSSION TITLE
Transforming Undetected Chromedriver into a Webdriver Compliant Server

### DISCUSSION THREAD

**Discussion by audioXD** [ultrafunkamsterdam/undetected-chromedriver#1275] [Ideas]
I've been reflecting on the potential of this project and its ability to benefit a wider community. Although undetected-chromedriver functions well, its utility seems largely confined to its original programming language. This can make it challenging to use in other languages, as highlighted in issue #1271, which mentioned a specific need for Java support.

Hence, I'm proposing a possible enhancement: transforming undetected-chromedriver into a WebDriver-compliant server, much like a sophisticated proxy. This transformation would allow the tool to manage launching, pathing, and sending intermittent CDP commands.

A key benefit of this change is the opportunity for seamless, transparent integration with a range of programming languages. In effect, we would expand the versatility and application of the tool, broadening its user base and potential uses.

Additionally, I suggest making use of the 'capabilities' page to manage complex configurations. By providing an intuitive interface for adjustments, users could tailor the tool to meet specific requirements and objectives more effectively.

I believe that this transformation could greatly expand the tool's impact and reach. I'd love to hear the community's thoughts on this idea and any potential challenges or further improvements that might be needed.

**Comment by DGolubets** [comment]
That would be great!
I have the same story like in the mentioned issue, but in a Rust project, using Thirtyfour.
I wish there was a Docker image compatible with Selenium protocol.
